tests/unit/dom/select-test.js

```diff
@@ -0,0 +1,210 @@
+    );
+  });
+
+  test('select with element that is disabled', async function (assert) {
+    element = buildInstrumentedElement('select');
+    element.disabled = true;
+    const option = 'example';
+
+    await setupContext(context);
+
+    assert.rejects(
+      select(element, option),
+      `Element is disabled when calling \`select(${element})\`./`
+    );
+  });
+
+  test('select | multiple: false - options.length > 1', async function (assert) {
+    element = buildInstrumentedElement('select');
+    element.multiple = false;
+    const options = ['apple', 'orange'];
+
+    await setupContext(context);
+
+    assert.rejects(
+      select(element, options),
+      `HTMLSelectElement \`multiple\` attribute is set to \`false\` but multiple options have been passed when calling \`select(${element})\``
+    );
+  });
+
+  test('select | 4 options - multiple: true - optionsToSelect.length : 2', async function (assert) {
+    const optionValues = ['apple', 'orange', 'pineapple', 'pear'];
+    element = buildInstrumentedElement('select');
+    element.multiple = true;
+
+    optionValues.forEach(optionValue => {
+      const optionElement = buildInstrumentedElement('option');
+      optionElement.value = optionValue;
+      element.append(optionElement);
+    });
+
+    const optionsToSelect = ['apple', 'orange'];
+
+    await setupContext(context);
+
+    await select(element, optionsToSelect);
+
+    assert.verifySteps(selectSteps);
+    assert.equal(element.selectedOptions.length, 2);
+    assert.equal(element.selectedOptions[0].value, 'apple');
+    assert.equal(element.selectedOptions[1].value, 'orange');
+  });
+
+  test('select | 4 options - multiple: false - optionsToSelect.length : 1', async function (assert) {
+    const optionValues = ['apple', 'orange', 'pineapple', 'pear'];
+    element = buildInstrumentedElement('select');
+    element.multiple = false;
+
+    optionValues.forEach(optionValue => {
+      const optionElement = buildInstrumentedElement('option');
+      optionElement.value = optionValue;
+      element.append(optionElement);
+    });
+
+    const optionsToSelect = 'apple';
+
+    await setupContext(context);
+
+    await select(element, optionsToSelect);
+
+    assert.verifySteps(selectSteps);
+    assert.equal(element.selectedIndex, 0);
+  });
+
+  test('select | multiple: false | select new option', async function (assert) {
+    const optionValues = ['apple', 'orange', 'pineapple', 'pear'];
+    element = buildInstrumentedElement('select');
+    element.multiple = false;
+
+    optionValues.forEach(optionValue => {
+      const optionElement = buildInstrumentedElement('option');
+      optionElement.value = optionValue;
+      element.append(optionElement);
+    });
+
+    await setupContext(context);
+
+    await select(element, 'apple');
+
+    await select(element, 'orange');
+
+    assert.verifySteps([...selectSteps, ...additionalSteps]);
```

Comment:
I didn't fully understand why we don't have to have `[..selectSteps, ...selectSteps]`. Specifically, what is avoiding firing `focus`/`focusin`.

Seems like this takes care of that:

https://github.com/emberjs/ember-test-helpers/blob/982320d5899af46f741bab4aea94468632b69cdc/addon-test-support/%40ember/test-helpers/dom/focus.ts#L18-L21

Figured I'd comment in case others were confused...

Reply:
Hmm my guess is because selecting doesnt actually remove focus so if it is focused still the event will run. You can see it in your demo if you dont click out and select something new.

Reply:
Yep, exactly. I was just surprised so I went digging to figure out why. 😃